en/docs/learn/api-security/oauth2/access-token-types/jwt-tokens.md

```diff
@@ -210,3 +209,58 @@ Follow the instructions below to import the public certificate into the client t
      ```
 
 5. Enter `wso2carbon` as the default password of the trust store when prompted.
+
+## Validating JWTs generated by external OAuth providers
+
+JWTs generated by an external OAuth provider can be validated by the gateway as per the following diagram.
+
+### Signature validation
+
+[![External JWT Signature Validation]({{base_path}}/assets/img/learn/api-security/oauth2/access-token-types/external-jwt-signature-validation.png)]({{base_path}}/assets/img/learn/api-security/oauth2/access-token-types/external-jwt-signature-validation.png)
+
+In order to do the signature validation of JWTs, you can use one of the following options.
+
+-  Import the public certificate into the client-truststore.jks located at `<API-M_HOME>/repository/resources/security/` with the **kid** value of the certificate as the alias.
```

Comment:
if the kid is not present in the jwt, it will validate with the certificate under gateway_certificate_alias . Please check on this as well.

Reply:
+1 fixed with https://github.com/wso2/docs-apim/pull/2637/commits/96a445fef7f6c7b7315ebbe30c6b431b94fd542c